prebuilt-tasks/src/main/java/com/redhat/parodos/tasks/jdbc/JdbcWorkFlowTask.java

```diff
@@ -0,0 +1,98 @@
+import java.util.List;
+import java.util.Map;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.redhat.parodos.workflow.exception.MissingParameterException;
+import com.redhat.parodos.workflow.parameter.WorkParameter;
+import com.redhat.parodos.workflow.parameter.WorkParameterType;
+import com.redhat.parodos.workflow.task.BaseWorkFlowTask;
+import com.redhat.parodos.workflow.task.enums.WorkFlowTaskOutput;
+import com.redhat.parodos.workflows.work.DefaultWorkReport;
+import com.redhat.parodos.workflows.work.WorkContext;
+import com.redhat.parodos.workflows.work.WorkReport;
+import com.redhat.parodos.workflows.work.WorkStatus;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class JdbcWorkFlowTask extends BaseWorkFlowTask {
+
+	enum OperationType {
+
+		QUERY, UPDATE, EXECUTE
+
+	}
+
+	private JdbcService service;
+
+	public JdbcWorkFlowTask() {
+		this.service = new JdbcServiceImpl();
+	}
+
+	JdbcWorkFlowTask(String beanName, JdbcService service) {
+		this.setBeanName(beanName);
+		this.service = service;
+	}
+
+	@Override
+	public @NonNull List<WorkParameter> getWorkFlowTaskParameters() {
+		LinkedList<WorkParameter> params = new LinkedList<>();
+		params.add(WorkParameter.builder().key("url").type(WorkParameterType.TEXT).optional(false).description(
+				"JDBC URL. E.g. jdbc:postgresql://localhost:5432/service?user=service&password=service123&ssl=true&sslmode=verify-full")
+				.build());
+		params.add(WorkParameter.builder().key("operation").type(WorkParameterType.TEXT).optional(false).description(
+				"Type of operation this statement is performing. One of " + Arrays.toString(OperationType.values()))
+				.build());
+		params.add(WorkParameter.builder().key("statement").type(WorkParameterType.TEXT).optional(false)
+				.description("The database statement to execute. E.g. 'select * from table'").build());
+		params.add(WorkParameter.builder().key("result-ctx-key").type(WorkParameterType.TEXT).optional(true)
+				.description("In query operation the result is stored in WorkContext with the provided key").build());
+		return params;
+	}
+
+	@Override
+	public @NonNull List<WorkFlowTaskOutput> getWorkFlowTaskOutputs() {
+		return List.of(WorkFlowTaskOutput.OTHER);
+	}
+
+	@Override
+	public WorkReport execute(WorkContext workContext) {
+		String url = "";
+		try {
+			url = getRequiredParameterValue("url");
+			String operation = getRequiredParameterValue("operation");
+			String statement = getRequiredParameterValue("statement");
+			String resultCtxKey = getOptionalParameterValue("result-ctx-key", "");
+
+			OperationType operationType = OperationType.valueOf(operation.toUpperCase());
+
+			switch (operationType) {
+
+				case QUERY -> {
+					List<Map<String, Object>> result = this.service.query(url, statement);
+					if (!resultCtxKey.isEmpty()) {
+						workContext.put(resultCtxKey, new ObjectMapper().writeValueAsString(result));
```

Comment:
The meaning of it, the result will be persisted to the worfklow-service db when the workflow-execution is updated, no?

Could you verify this? Not sure it is desired to store the information on the context in the DB, or at least, let there be an option if it should or not.

Reply:
we put the result in the context. this is the same for all our tasks that have a result. Like get, select, and such. This is a requirement by Piotr and Luke. If you have a different suggestion then please talk to them.